client/my-sites/controller.js

```diff
@@ -553,9 +553,15 @@ export function siteSelection( context, next ) {
 		// onSelectedSiteAvailable might render an error page about domain-only sites or redirect
 		// to wp-admin. In that case, don't continue handling the route.
 		dispatch( setSelectedSiteId( siteId ) );
-		if ( onSelectedSiteAvailable( context ) ) {
-			next();
-		}
+
+		// This will fetch the site and update the state after the plan is upgraded
+		dispatch( requestSite( siteFragment ) )
```

Comment:
I think this is not the right place to add this request, as it will trigger a new API request _every time a user navigates to a site-specific page_.

I think we need to work out if we can invalidate the site cache when a plan purchase occurs.

Reply:
Thanks for catching that! I wasn't aware this was triggered on every site-specific navigation.